src/relax/ir/block_builder.cc

```diff
@@ -136,10 +136,18 @@ class BlockBuilderNode::ExprNormalizer : public ExprFunctor<Expr(const Expr&)> {
     }
 
     // only do shape/type inference if the Call does not have shape/type
-    if (call->shape_ && call->checked_type_.defined()) {
+    if (call->shape_.defined() && call->checked_type_.defined()) {
```

Comment:
is `defined()` necessary given shape_ is Optional?

Reply:
No it’s not necessary. Just because I would always prefer explicitly using `.defined()` for more clarity.